### PR TITLE
Resolving conflicts in make upgrade

### PR DESCRIPTION
**Pull request by brittneyexline**
**Description:**
tl;dr, we should find a way to stop pinning as many dependencies as we can to avoid update conflicts with edx-platform.

Longer:
I started by running make upgrade in edx-platform against this branch, and adding things that failed as a constraint in requirements/test-master.in, and then running make upgrade in edx-enterprise, then repeating the process. There were several issues going on, but the main problem is that we are trying to stay in sync with edx-platform, but we pin explicitly when a package is upgraded, and it may be constrained for some reason in edx-platform. It took a while to come to this, and there is a corresponding change necessary in edx-platform because of transifex-client, which we would have had to resolve anyway in edx-platform.

Here is the corresponding edx-platform branch I used to test make upgrade: https://github.com/edx/edx-platform/pull/21132

**Merge deadline:** ASAP - we are blocking make upgrade from being run without errors in edx-platform.

**Merge checklist:**

- [ ] New requirements are in the right place
    - `base.in` if needed in production, but edx-platform doesn't install it.
    - `test-master.in` if edx-platform pins it, with a matching version.
- [ ] Regenerate requirements with `make upgrade && make requirements` (and make sure to fix any errors).
  **DO NOT** just add dependencies to `requirements/*.txt` files.
- [ ] Called `make static` for webpack bundling if any static content was updated.
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are (reasonably) squashed
- [ ] Translations are updated
- [ ] PR author is listed in AUTHORS

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPi after tag-triggered build is finished.
- [ ] Delete working branch (if not needed anymore)
- [ ] edx-platform PR (be sure to include edx-platform requirements upgrades that were present in this PR)

**Author concerns:** List any concerns about this PR - inelegant 
solutions, hacks, quick-and-dirty implementations, concerns about 
migrations, etc.

### CODE REVIEW

**Comment by brittneyexline** [comment]
@muhammad-ammar @mushtaqak @irfanuddinahmad - please feel free to merge and try upgrading in edx-platform when you come online, hopefully everything should work 🤞
